### PR TITLE
processor(ticdc): add sourceManager and sinkManager into processor

### PR DESCRIPTION
**Pull request by 0xPoe**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/5928

### What is changed and how it works?

add sourceManager and sinkManager into processor.

#### Tests 

 - Unit test

##### Will it cause performance regression or break compatibility?
No
##### Do you need to update user documentation, design documentation or monitoring documentation?
No
### Release note 

```release-note
None
```

### CODE REVIEW

**Review by nongfushanquan — APPROVED** [review]

File: pkg/config/debug.go [hunk 1]
```diff
@@ -60,3 +60,8 @@ func (c *DebugConfig) ValidateAndAdjust() error {
 	}
 	return nil
 }
+
+// IsPullBasedSinkEnabled returns whether pull-based sink is enabled.
+func (c *DebugConfig) IsPullBasedSinkEnabled() bool {
+	return c.EnablePullBasedSink && c.EnableDBSorter
```
Comment (hicqu) [inline pkg/config/debug.go:66]: Please left a comment that we will transform Unified Sorter into SortEngine.

**Review by hicqu — APPROVED** [review]

**Pull request merged**